### PR TITLE
[FrameworkBundle] Move profiling configuration to PHP

### PR DESCRIPTION
**Pull request by hvt**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | #37186 
| License       | MIT

Moving profiling configuration in framework bundle to PHP.

### CODE REVIEW

**Comment by hvt** [comment]
Failures seem unrelated to me, but not sure...

**Review by javiereguiluz — APPROVED** [review]
It looks OK to me!

I only have a minor question: to be exactly the same as the XML config, we should add this too:

```php
$container->services()
    ->defaults()->private()

    ->set('profiler', Profiler::class)
    // ...
```

@Tobion do you think we should add this explicitly or rely on the "services are private by default" feature?

**Comment by Tobion** [comment]
> rely on the "services are private by default"

This is it. The other PRs all didn't add `->defaults()->private()` as it's unneeded.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/profiling.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('profiler', Profiler::class)
+            ->public()
+            ->args([service('profiler.storage'), service('logger')->nullOnInvalid()])
+            ->tag('monolog.logger', ['channel' => 'profiler'])
+
+        ->set('profiler.storage', FileProfilerStorage::class)
+            ->args([param('profiler.storage.dsn')])
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/profiling.php:26]: please, update and rebase with master branch to get the missing `param` function:
https://travis-ci.org/github/symfony/symfony/jobs/697135256#L3705

Comment (hvt) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/profiling.php:26]: Done :].

**Review by yceruto — APPROVED** [review]
LGTM, thanks!

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @hvt.

**Comment by stof** [comment]
> I only have a minor question: to be exactly the same as the XML config, we should add this too:

The XML config files have added these defaults in the 3.x era, when it was not the default yet.

**Pull request merged**